etcdctl/command/util.go

```diff
@@ -221,6 +221,9 @@ func mustNewClient(c *cli.Context) client.Client {
 		err := hc.Sync(ctx)
 		cancel()
 		if err != nil {
+			if err == client.ErrNoEndpoints {
+				fmt.Fprintf(os.Stderr, "Got empty client urls from member list when synchronizing through %s.\n", strings.Join(eps, ","))
```

Comment:
etcd cluster has no published client endpoints.
Try `--no-sync` if you want to access a non-published(%v) client endpoint.